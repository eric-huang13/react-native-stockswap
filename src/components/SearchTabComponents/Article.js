import React, {Component} from 'react';
import {Text, SafeAreaView, View, Image, StyleSheet} from 'react-native';
import {moderateScale} from '../../util/responsiveFont';

export class Article extends Component {
  render() {
    const {item} = this.props;
    console.log(this.props, 'props in news')

    return (
      <SafeAreaView style={style.container}>
        <View style={style.imageContainer}>
          <Image style={style.image} source={{uri: item.image_url}} />
        </View>
        <View style={style.headlineContainer}>
          <Text style={style.headline}>{item.title}</Text>
          {/* <Text style={style.headline}>{item.news_url}</Text> */}

          <Text style={style.intro}>{item.text}</Text>
        </View>
        <View style={style.sourceContainer}>
          <Text style={style.source}>{item.source_name}</Text>
          <Text style={style.released}>{item.date}</Text>
        </View>
      </SafeAreaView>
    );
  }
}

export default Article;

const style = StyleSheet.create({
  container: {
    marginTop: moderateScale(4),
    margin: moderateScale(8),
    paddingBottom: moderateScale(8),
    backgroundColor: '#324165',
    borderRadius: moderateScale(4),
  },
  sourceContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: moderateScale(6),
  },
  source: {
    color: '#9ea6b5',
    marginRight: moderateScale(4),
    fontSize: moderateScale(15),
    fontFamily: 'Montserrat-Regular',
  },
  released: {
    color: '#9ea6b5',
    fontSize: moderateScale(13),
    fontFamily: 'Montserrat-Regular',
  },
  headlineContainer: {
    flexDirection: 'column',
    paddingHorizontal: moderateScale(6),
    marginBottom: moderateScale(6),
  },
  headline: {
    color: '#FFFFFF',
    fontSize: moderateScale(15),
    marginTop: moderateScale(8.5),
    marginBottom: moderateScale(10),
    fontFamily: 'Montserrat-Bold',
  },
  intro: {
    color: '#FFFFFF',
    fontSize: moderateScale(15),
    fontFamily: 'Montserrat-Medium',
  },
  imageContainer: {
    // paddingLeft: 3,
    // paddingRight: 3,
  },
  image: {
    borderTopLeftRadius: moderateScale(4),
    borderTopRightRadius: moderateScale(4),
    height: moderateScale(140),
    marginLeft: moderateScale(0.05),
    marginRight: moderateScale(0.05),
    width: '100%',
  },
});
