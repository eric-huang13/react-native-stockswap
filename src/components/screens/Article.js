import React, {Component} from 'react';
import {Text, View, Image, StyleSheet} from 'react-native';

export class Article extends Component {
  render() {
    const {item} = this.props;

    return (
      <View style={style.container}>
        <View style={style.imageContainer}>
          <Image style={style.image} source={{uri: item.img}} />
        </View> 
        <View style={style.headlineContainer}>      
        <Text style={style.headline}>{item.headline}</Text>
        <Text style={style.intro}>{item.intro}</Text>

        </View> 
        <View style={style.sourceContainer}>
          <Text style={style.source}>{item.source}</Text>
          <Text style={style.released}>{item.released} ago</Text>
        </View>
      </View>
    );
  }
}

export default Article;

const style = StyleSheet.create({
  container: {
    marginTop: 1.2,
    margin: 15,
    paddingBottom: 8,
    backgroundColor: '#324165',
    borderRadius:4
    },
  sourceContainer: {
    flexDirection: 'row',
    justifyContent:'space-between',
    paddingHorizontal:6,
  },
  source: {
    color: 'grey',
    marginRight: 4,
    fontSize: 16,
  },
  released: {
    color: 'grey',
    fontSize: 14,
  },
  headlineContainer:{
    flexDirection:'column',
    paddingHorizontal:6,
    marginBottom: 6,

  },
  headline: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
    marginTop: 8.5,
    marginBottom: 10,
  },
  intro:{
    color: 'white',
    fontSize: 16,
  },
  imageContainer: {
    // paddingLeft: 3,
    // paddingRight: 3,

  },
  image: {
    borderTopLeftRadius:6,
    borderTopRightRadius:6,
    height: 140,
    marginLeft: 0.05,
    marginRight: 0.05,
    width: '100%',
  },
});
