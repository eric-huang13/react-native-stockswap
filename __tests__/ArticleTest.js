import React from 'react';
import Article from '../src/components/SearchTabComponents/Article';
import renderer from 'react-test-renderer';


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
    const tree = renderer.create(<Article item={[]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});