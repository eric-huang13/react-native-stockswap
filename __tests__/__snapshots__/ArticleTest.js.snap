// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#324165",
      "borderRadius": 5.649635036496351,
      "margin": 11.299270072992702,
      "marginTop": 5.649635036496351,
      "paddingBottom": 11.299270072992702,
    }
  }
>
  <View
    style={Object {}}
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderTopLeftRadius": 5.649635036496351,
          "borderTopRightRadius": 5.649635036496351,
          "height": 197.73722627737226,
          "marginLeft": 0.07062043795620437,
          "marginRight": 0.07062043795620437,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "marginBottom": 8.474452554744525,
        "paddingHorizontal": 8.474452554744525,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 21.186131386861312,
          "marginBottom": 14.124087591240876,
          "marginTop": 12.005474452554743,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 21.186131386861312,
          "marginBottom": 14.124087591240876,
          "marginTop": 12.005474452554743,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 21.186131386861312,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 8.474452554744525,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9ea6b5",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 21.186131386861312,
          "marginRight": 5.649635036496351,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#9ea6b5",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 18.361313868613138,
        }
      }
    >
       ago
    </Text>
  </View>
</RCTSafeAreaView>
`;
